modules/xfem/include/efa/EFAFuncs.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef EFAFUNCS_H
+#define EFAFUNCS_H
+
+#include <map>
+#include <set>
+#include <vector>
+#include <algorithm>
+
+template <typename T>
+bool deleteFromMap(std::map<unsigned int, T*> &theMap, T* elemToDelete)
+{
+  bool didIt = false;
+  typename std::map<unsigned int, T*>::iterator i=theMap.find(elemToDelete->id());
+  if (i != theMap.end())
+  {
+    delete i->second;
+    theMap.erase(i);
```

Comment:
Are you ever going to pull this code back out? If so you could add your own shared pointer wrapper. You could also simply require C++11 (we are going that route sooner or later anyways)!
The compelling reason here is that for such a complex piece of code the probability you are going to leak memory sooner or later is pretty big.